how-tos/staging-release-setup.md

```diff
@@ -0,0 +1,83 @@
+python archiver_client.py mozharness --destination mozharness-central --repo mozilla-central --rev default --debug  # Central must be used against every branch
+```
+
+2. Modify the `mozharness-central/configs/merge_day/central_to_beta.py` file to look like this
+```diff
+--- mozharness-central.orig/configs/merge_day/central_to_beta.py2017-07-10 06:26:15.000000000 -0400
++++ mozharness-central/configs/merge_day/central_to_beta.py20172017-07-10 09:53:01.838515927 -0400
+@@ -67,8 +67,8 @@
+     # "hg_share_base": None,
+     "tools_repo_url": "https://hg.mozilla.org/build/tools",
+     "tools_repo_branch": "default",
+-    "from_repo_url": "ssh://hg.mozilla.org/mozilla-central",
+-    "to_repo_url": "ssh://hg.mozilla.org/releases/mozilla-beta",
++    "from_repo_url": "ssh://hg.mozilla.org/projects/date",
++    "to_repo_url": "ssh://hg.mozilla.org/projects/jamun",
+
+     "base_tag": "FIREFOX_BETA_%(major_version)s_BASE",
+     "end_tag": "FIREFOX_BETA_%(major_version)s_END",
+```
+
+3. Run the actual script to get the "diff" between the two branches:
+```sh
+python mozharness-central/scripts/merge_day/gecko_migration.py -c merge_day/central_to_beta.py
+```
+
+4. The script should have created a diff. Check that everything is okay and push to jamun
+```sh
+hg -R build/jamun diff
+hg out -r . jamun
+hg push
+```
+
+## Staging tools
+
+- staging [Ship-it](https://ship-it-dev.allizom.org/)
+- staging [release runner](https://hg.mozilla.org/build/puppet/file/default/manifests/moco-nodes.pp#l633)
+- release automation notificats [group](https://groups.google.com/a/mozilla.com/forum/?hl=en#!forum/release-automation-notifications-dev) and `#release-notifications-dev` IRC channel
+- staging [balrog](https://balrog-admin.stage.mozaws.net/)
+
+A note on Balrog. Historically, we've had issues with plugging the staging instance to
+the staging release pipeline, so we used another workaround:
+- from aws console, create a new ubuntu AWS instance
+- login to it and clone Balrog [codebase](https://github.com/mozilla/balrog)
+- follow the [installation](https://github.com/mozilla/balrog#installation) process
+- point that instance in all configs [mozharnss configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) that need it. Sample [here](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/dev_updates_firefox_beta.py#23)
+
+## Staging configs
+
+- release runner will consume [project branch configs](https://dxr.mozilla.org/build-central/source/buildbot-configs/mozilla/project_branches.py#114)
+- mozharness [configs](https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/configs/releases/) - all configs with `dev` prefix
+- in order to avoid pushing patcher configs changes/tags to *real* [tools](http://hg.mozilla.org/build/tools/) repo, please point to a user fork. You can reuse rail's [tools repo](https://hg.mozilla.org/users/raliiev_mozilla.com/tools)
```

Comment:
Nit: rail's link is broken.

After you left, rail showed me how to quickly create a custom repo.
```sh
ssh hg.mozilla.org clone tools-fake build/tools
# Will show: Please wait.  Cloning /build/tools to /users/jlorenzo_mozilla.com/tools-fake
```
I guess it's just better to reclone a repo from there. That'll avoid missing permissions.


Source: https://wiki.mozilla.org/Release:Release_Automation_on_Mercurial:Staging_Specific_Notes#Setup_staging_repos